ci/fireci/fireciplugins/macrobenchmark.py

```diff
@@ -0,0 +1,206 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+import glob
+import json
+import logging
+import os
+import random
+import sys
+
+import pystache
+import shutil
+import yaml
+
+from fireci import ci_command
+from fireci.dir_utils import chdir
+
+_logger = logging.getLogger('fireci.macrobenchmark')
+
+
+@ci_command()
+def macrobenchmark():
+  """Measures app startup times for Firebase SDKs."""
+  asyncio.run(_launch_macrobenchmark_test())
+
+
+async def _launch_macrobenchmark_test():
+  _logger.info('Starting macrobenchmark test...')
+
+  artifact_versions, config, _ = await asyncio.gather(
+    _parse_artifact_versions(),
+    _parse_config_yaml(),
+    _create_gradle_wrapper()
+  )
+
+  with chdir('macrobenchmark'):
+    runners = [MacrobenchmarkTest(k, v, artifact_versions) for k, v in config.items()]
+    await asyncio.gather(*[x.run() for x in runners])
+
+  _logger.info('Macrobenchmark test finished.')
+
+
+async def _parse_artifact_versions():
+  proc = await asyncio.subprocess.create_subprocess_exec('./gradlew', 'assembleAllForSmokeTests')
+  await proc.wait()
+
+  with open('build/m2repository/changed-artifacts.json', 'r') as json_file:
+    artifacts = json.load(json_file)
+  return dict(_artifact_key_version(x) for x in artifacts['headGit'])
+
+
+def _artifact_key_version(artifact):
+  group_id, artifact_id, version = artifact.split(':')
+  return f'{group_id}:{artifact_id}', version
+
+
+async def _parse_config_yaml():
+  with open('macrobenchmark/config.yaml', 'r') as yaml_file:
+    return yaml.safe_load(yaml_file)
+
+
+async def _create_gradle_wrapper():
+  with open('macrobenchmark/settings.gradle', 'w'):
+    pass
```

Comment:
is this intended?

Reply:
Yes. Without a `settings.gradle` file, `./gradlew wrapper` failed with this error:
```
FAILURE: Build failed with an exception.

* What went wrong:
Project directory '/Users/yifany/Documents/firebase/firebase-android-sdk/macrobenchmark' is not part of the build defined by settings file '/Users/yifany/Documents/firebase/firebase-android-sdk/settings.gradle'. If this is an unrelated build, it must have its own settings file.
```
And looks like an empty `settings.gradle` is good enough.